ledger/store/trackerdb/generickv/schema.go

```diff
@@ -23,149 +23,378 @@ import (
 )
 
 const (
-	kvPrefixAccount              = "account"
-	kvPrefixResource             = "resource"
-	kvPrefixAppKv                = "appkv"
-	kvPrefixCreatorIndex         = "creator"
-	kvPrefixOnlineAccount        = "online_account_base"
-	kvPrefixOnlineAccountBalance = "online_account_balance"
-	kvRoundKey                   = "global_round"
-	kvSchemaVersionKey           = "global_schema_version"
-	kvTotalsKey                  = "global_total"
-	kvTxTail                     = "txtail"
-	kvOnlineAccountRoundParams   = "online_account_round_params"
-	kvPrefixStateproof           = "stateproofs"
+	prefixLength    = 2
+	separatorLength = 1
+	addressLength   = 32
+	roundLength     = 8
+)
+
+const (
+	kvPrefixAccount              = "xa"
+	kvPrefixResource             = "xb"
+	kvPrefixAppKv                = "xc"
+	kvPrefixCreatorIndex         = "xd"
+	kvPrefixOnlineAccount        = "xe"
+	kvPrefixOnlineAccountBalance = "xf"
+	kvRoundKey                   = "xg"
+	kvSchemaVersionKey           = "xh"
+	kvTotalsKey                  = "xi"
+	kvTxTail                     = "xj"
+	kvOnlineAccountRoundParams   = "xk"
+	kvPrefixStateproof           = "xl"
+)
+
+const (
+	// this is the true separator used in the keys
+	separator = '-'
+	// this is used as a value greather than the `separator` to get all the keys with a given prefix
+	endRangeSeparator = '.'
 )
 
 // return the big-endian binary encoding of a uint64
-func bigEndianUint64(v uint64) []byte {
-	ret := make([]byte, 8)
-	binary.BigEndian.PutUint64(ret, v)
+func bigEndianUint64(v uint64) [8]byte {
```

Comment:
does it really helped? what is `go build -gcflags "-m=3" ./ledger/store/trackerdb/generickv 2>&1 | grep schema.go` output about heap alloc?

Reply:
it says everything can be inlined and nothing escapes, except for 
```
ledger/store/trackerdb/generickv/schema.go:115:13: make([]byte, 0, prefixLength + separatorLength + len(kvKey)) escapes to heap
```

The kvKey arg is variable length, so it escapes it to the heap, i don't know how to force it to the stack if it's small enough.